libraries/AP_OSD/AP_OSD_ParamScreen.cpp

```diff
@@ -0,0 +1,710 @@
+        p->copy_name_token(setting._current_token, name, 16);
+        name[16] = 0;
+
+        const AP_OSD_ParamSetting::ParamMetadata* metadata = setting.get_custom_metadata();
+
+        uint16_t value_pos = 19;
+        backend->write(x, y, param_blink, "%s:", name);
+
+        switch (type) {
+        case AP_PARAM_INT8: {
+            int8_t val = ((AP_Int8*)p)->get();
+            if (metadata != nullptr && val >= 0 && val < metadata->values_max) {
+                backend->write(value_pos, y, value_blink, "%s", metadata->values[val]);
+            } else {
+                backend->write(value_pos, y, value_blink, "%hhd", val);
+            }
+        }
+            break;
+        case AP_PARAM_INT16: {
+            int16_t val = ((AP_Int16*)p)->get();
+            if (metadata != nullptr && val >= 0 && val < metadata->values_max) {
+                backend->write(value_pos, y, value_blink, "%s", metadata->values[val]);
+            } else {
+                backend->write(value_pos, y, value_blink, "%hd", val);
+            }
+        }
+            break;
+        case AP_PARAM_INT32: {
+           int32_t val = ((AP_Int16*)p)->get();
+            if (metadata != nullptr && val >= 0 && val < metadata->values_max) {
+                backend->write(value_pos, y, value_blink, "%s", metadata->values[val]);
+            } else {
+                backend->write(value_pos, y, value_blink, "%d", val);
+            }
+        }
+            break;
+        case AP_PARAM_FLOAT: {
+            const float val = ((AP_Float*)p)->get();
+            // cope with really small value
+            if (val < 0.01 && !is_zero(val)) {
+                backend->write(value_pos, y, value_blink, "%.4f", val);
+            } else if (val < 0.001 && !is_zero(val)) {
+                backend->write(value_pos, y, value_blink, "%.5f", val);
+            } else {
+                backend->write(value_pos, y, value_blink, "%.3f", val);
+            }
+        }
+            break;
+        case AP_PARAM_VECTOR3F:
+        case AP_PARAM_NONE:
+        case AP_PARAM_GROUP:
+            break;
+        }
+    }
+}
+
+// modify the selected parameter number
+void AP_OSD_ParamScreen::modify_parameter(uint8_t number, Event ev)
+{
+    if (number > NUM_PARAMS) {
+        return;
+    }
+
+    const AP_OSD_ParamSetting& setting = params[number-1];
+    AP_Param* p = setting._param;
+
+    if (p->is_read_only()) {
+        return;
+    }
+
+    _requires_save |= 1 << (number-1);
+
+    float incr = setting._param_incr * ((ev == Event::MENU_DOWN) ? -1.0f : 1.0f);
+    int32_t incr_int = int32_t(roundf(incr));
+    int32_t max_int = int32_t(roundf(setting._param_max));
+    int32_t min_int = int32_t(roundf(setting._param_min));
+
+    if (p != nullptr) {
+        switch (setting._param_type) {
+        // there is no way to validate the ranges, so as a rough guess prevent
+        // integer types going below -1;
+        case AP_PARAM_INT8: 
+        {
```

Comment:
Atypical formatting; usually the break goes inside the {} and the { would be on the same line as the colon on the case statement.

Reply:
Fixed - and all the other places as well